src/sched.rs

```diff
@@ -1,204 +1,127 @@
+        CLONE_VM,
+        CLONE_FS,
+        CLONE_FILES,
+        CLONE_SIGHAND,
+        CLONE_PTRACE,
+        CLONE_VFORK,
+        CLONE_PARENT,
+        CLONE_THREAD,
+        CLONE_NEWNS,
+        CLONE_SYSVSEM,
+        CLONE_SETTLS,
+        CLONE_PARENT_SETTID,
+        CLONE_CHILD_CLEARTID,
+        CLONE_DETACHED,
+        CLONE_UNTRACED,
+        CLONE_CHILD_SETTID,
+        CLONE_NEWUTS,
+        CLONE_NEWIPC,
+        CLONE_NEWUSER,
+        CLONE_NEWPID,
+        CLONE_NEWNET,
+        CLONE_IO,
     }
 }
 
-// Support a maximum CPU set of 1024 nodes
-#[cfg(all(target_arch = "x86_64", target_os = "linux"))]
+#[cfg(target_pointer_width = "64")]
 mod cpuset_attribs {
-    use super::CpuMask;
-    pub const CPU_SETSIZE:           usize = 1024;
-    pub const CPU_MASK_BITS:         usize = 64;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u64 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u64 << bit)
-    }
+    pub const CPU_MASK_BITS: usize = 64;
+    pub const FLAG_BIT: u64 = 1;
 }
 
-#[cfg(all(target_arch = "x86", target_os = "linux"))]
+#[cfg(target_pointer_width = "32")]
 mod cpuset_attribs {
-    use super::CpuMask;
-    pub const CPU_SETSIZE:           usize = 1024;
-    pub const CPU_MASK_BITS:         usize = 32;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u32 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u32 << bit)
-    }
+    pub const CPU_MASK_BITS: usize = 32;
+    pub const FLAG_BIT: u32 = 1;
 }
 
-#[cfg(all(target_arch = "aarch64", any(target_os = "linux", target_os = "android")))]
-mod cpuset_attribs {
-    use super::CpuMask;
-    pub const CPU_SETSIZE:           usize = 1024;
-    pub const CPU_MASK_BITS:         usize = 64;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u64 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u64 << bit)
-    }
+#[inline]
+fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
+    cur | (cpuset_attribs::FLAG_BIT << bit)
 }
 
-#[cfg(all(any(target_arch = "arm", target_arch = "mips"), target_os = "android"))]
-mod cpuset_attribs {
-    use super::CpuMask;
-    // bionic only supports up to 32 independent CPUs, instead of 1024.
-    pub const CPU_SETSIZE:          usize = 32;
-    pub const CPU_MASK_BITS:        usize = 32;
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u32 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u32 << bit)
-    }
+#[inline]
+fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
+    cur & !(cpuset_attribs::FLAG_BIT << bit)
```

Comment:
I'm not sure these helpers are really necessary given the pretty singular purpose of the code at the call site and the fact that these are only called once.